analyses/hotspots-detection/01-create-hotspot-maf.Rmd

```diff
@@ -0,0 +1,185 @@
+unique_cols <- c("t_depth",
+                 "n_depth",
+                 "t_ref_count",
+                 "n_ref_count",
+                 "t_alt_count",
+                 "n_alt_count",
+                 "caller",
+                 "vcf_qual")
+
+# Gather mean values in `unique_cols` such that 
+# calls from multiple callers are summarised as 1 unique row
+combined_maf_hotspots_filt <- combined_maf_hotspots_filt %>%
+  group_by_at(setdiff(names(combined_hotspot_calls), unique_cols)) %>%
+  summarise(
+            t_depth= mean(as.numeric(t_depth)),
+            n_depth = mean(as.numeric(n_depth)),
+            t_ref_count = mean(as.numeric(t_ref_count)),
+            n_ref_count = mean(as.numeric(n_ref_count)),
+            t_alt_count = mean(as.numeric(t_alt_count)),
+            n_alt_count = mean(as.numeric(n_alt_count)),
+            caller_count = n(),
+            caller = toString(caller),
+            vcf_qual = mean(as.numeric(vcf_qual))) %>% 
+  ungroup() %>%
+  # calculate VAF
+  mutate(VAF = t_alt_count/(t_alt_count+t_ref_count))
+
+```
+
+## Mean read support for combination of callers
+
+We will try to identify the read support for each combination of callers to visualize if any callers call hostspot sites with low read support calls which
+```{r}
+
+ggplot(combined_maf_hotspots_filt[,c("caller","caller_count","t_alt_count")],
+       aes(x=as.factor(caller_count),y=t_alt_count))+
+  geom_violin()+ 
+  ggpubr::stat_compare_means()
+
+```
+Hotspot sites called by just 1 caller seem to have majority low read support and a few are found in 3 caller ( now captured in 3 callers because of the variant being present in vardict)
+
+Lets look at these calls that are unique to 1 caller.
+
+```{r}
+
+unique_1caller_hotspots <- combined_maf_hotspots_filt %>%
+  filter(caller_count==1)
+
+unique_1caller_hotspots
+```
+There are 33 calls that are only called by 1 caller.
+
+What's the distrbution of the t_alt_count in calls only supported by 1 caller?
+```{r}
+
+summary.caller <- unique_1caller_hotspots %>%  
+  group_by(caller) %>%
+  tally()
+
+
+ggplot(unique_1caller_hotspots[,c("caller","t_alt_count")],
+       aes(x=as.factor(caller),y=t_alt_count))+
+  geom_violin()+ 
+  geom_jitter(height = 0, width = 0.1) +
+  # add number of calls at the top
+  geom_text(data=summary.caller ,aes(x = caller, y = 350, label=n), fontface =2, size = 4)+
+  # draw a line where t_alt_count==5
+  geom_hline(aes(yintercept=5,color="red"))+
+  theme(legend.position="none")
+
+
+```
+
+lancet and vardict have majority of the unique calls. However, some calls from lancet,mutect2 and vardict seem to be below the red line which denotes t_alt_count==5.
+
+## Merge the combined hotspots with consensus 
+
+We will now create a maf file with hotspots called uniquely from `combined_hotspot_calls` to `consensus_maf_file`
+```{r}
+
+combined_maf_hotspots_filt %>%
+  bind_rows(consensus_maf_file) %>%
+  write_tsv(gzfile(output_file),na = ".")
```

Comment:
I think (unless this is a new feature) that write_tsv handles compression automatically.
```suggestion
  write_tsv(output_file, na = ".")
```